development/frontend/testing.md

```diff
@@ -0,0 +1,73 @@
+## Front-End Testing
+
+For our front-end we test on two fronts:
+
+- [Unit Tests](#unit-tests) - Focusses on individual functional testing. Each function should be tested in complete isolation
+- [E2E Integration Tests](#e2e-integration-tests) - Driven by automated testing framework, [Cypress](https://www.cypress.io/), this tests our front-end in it's entirety, including button clicks, navigations and API calls.
+
+## Unit Tests
+
+Not yet written
+
+## E2E Integration Tests
+
+### Running Locally
+
+To run the E2E tests yourself, we need to first run our own local web server that the tests will run on. For our testing, we have a single server configured with multiple users, teams and projects for testing purposes. To run this, open a terminal and run:
+
+```bash
+npm run cy:web-server
+```
+
+Once this is up and running, you then have two options:
+
+#### Run Tests via Terminal
+
+```bash
+npm run cy:run
+```
+
+Simply run, in a terminal, all of the front-end E2E tests and displays the results. If you just want to run all the tests, and aren't necessarily developing your own, this is the command to run.
+
+
+#### Run Tests via Cypress Application
+
+```bash
+npm run cy:open
+```
+
+Opens the Cypress application. From here, you can run individual test files, and see the framework clicking through the UI as the tests are run. If you're developing or debugging your own tests, we recommend this. 
+
+### Hints & Tips
+
+If you're looking to write your own Cypress tests, then here are some tips that can help you get started:
+
+#### beforeEach
+
+If you explore some of the existing tests, you'll see that `beforeEach` is a very useful way of ensuring consistent behaviour across a group of tests, e.g. starting from the same point, or making sure a user is logged in.
+
+#### Helper Functions
+
+To make Cypress testing easier, we have created a collection of helper functions that can be called within any loop of the testing framework.
+
+```js
+cy.login(<username>, <password>)
+```
+This will log you into FlowForge as a given user. Our users are defined in the [test environment](https://github.com/flowforge/flowforge/blob/main/test/e2e/frontend/environments/standard.js).
```

Comment:
```suggestion
This will log you into FlowForge as the given user. The test users are defined in the [test environment](https://github.com/flowforge/flowforge/blob/main/test/e2e/frontend/environments/standard.js).
```